linkerd/app/outbound/src/tests/tcp.rs

```diff
@@ -317,6 +325,83 @@ async fn load_balances() {
     );
 }
 
+#[tokio::test]
+async fn no_profiles_when_outside_search_nets() {
+    let _trace = test_support::trace_init();
+
+    let profile_addr = SocketAddr::new([10, 0, 0, 42].into(), 5550);
+    let no_profile_addr = SocketAddr::new([126, 32, 5, 18].into(), 5550);
+    let cfg = Config {
+        allow_discovery: IpMatch::new(Some(IpNet::from_str("10.0.0.0/8").unwrap())),
+        ..default_config(profile_addr)
+    };
+    let id_name = linkerd2_identity::Name::from_hostname(
+        b"foo.ns1.serviceaccount.identity.linkerd.cluster.local",
+    )
+    .expect("hostname is valid");
```

Comment:
Think this should be:

```suggestion
    .expect("hostname is invalid");
```